### PR TITLE
Added new variables for logging role for remote-syslog plugin

### PR DESCRIPTION
**Pull request by sqtran**
This commit adds a few new parameters useful for the fluent-plugin-remote-syslog work.  These won't be useful until https://github.com/openshift/origin-aggregated-logging/pull/679 is merged into the logging project first.

### CODE REVIEW

File: roles/openshift_logging_fluentd/templates/fluentd.j2 [hunk 1]
```diff
@@ -120,6 +120,52 @@ spec:
         - name: "MUX_CLIENT_MODE"
           value: "{{ openshift_logging_mux_client_mode }}"
 {% endif %}
+
+{% if openshift_logging_fluentd_remote_syslog is defined %}
```
Comment (richm) [inline roles/openshift_logging_fluentd/templates/fluentd.j2]:
```
 +{% if openshift_logging_fluentd_remote_syslog is defined and openshift_logging_fluentd_remote_syslog %} 
```
I think that's how you say "if openshift_logging_fluentd_remote_syslog is defined and openshift_logging_fluentd_remote_syslog is True"

File: roles/openshift_logging_mux/templates/mux.j2 [hunk 2]
```diff
@@ -119,6 +119,52 @@ spec:
               resource: limits.memory
         - name: "FILE_BUFFER_LIMIT"
           value: "{{ openshift_logging_mux_file_buffer_limit | default('2Gi') }}"
+
+{% if openshift_logging_fluentd_remote_syslog is defined %}
```
Comment (richm) [inline roles/openshift_logging_mux/templates/mux.j2]: see comment above

File: roles/openshift_logging/README.md [hunk 3]
```diff
@@ -194,3 +194,14 @@ Elasticsearch OPS too, if using an OPS cluster:
...
+### remote syslog forwarding
+`openshift_logging_fluentd_remote_syslog`: Set true to enable remote syslog forwarding, defaults to false
+`openshift_logging_fluentd_remote_syslog_host`: Required, hostname or IP of remote syslog
+`openshift_logging_fluentd_remote_syslog_port`: Port of remote syslog server, defaults to 514
+`openshift_logging_fluentd_remote_syslog_severity`: Severity level, defaults to debug
+`openshift_logging_fluentd_remote_syslog_facility`: Syslog facility, defaults to local0
+`openshift_logging_fluentd_remote_syslog_remove_tag_prefix`: Remove the prefix from the tag
+`openshift_logging_fluentd_remote_syslog_tag_key`: Use the field specified to set the key field in the syslog message
+`openshift_logging_fluentd_remote_syslog_use_record`: Use severity and facility from record if available
+`openshift_logging_fluentd_remote_syslog_payload_key`: Use the field specified from record to set payload
```
Comment (richm) [inline roles/openshift_logging/README.md]: Is `openshift_logging_fluentd_remote_syslog_payload_key` a boolean or a string value?

Comment (richm) [reply roles/openshift_logging/README.md]: Users need to be able to configure fluentd and mux separately, so there should be `openshift_logging_mux_*` versions of these parameters that are used in `mux.j2`

File: roles/openshift_logging/README.md [hunk 4]
```diff
@@ -199,3 +199,14 @@ Elasticsearch OPS too, if using an OPS cluster:
...
+### remote syslog forwarding
+`openshift_logging_fluentd_remote_syslog`: Set true to enable remote syslog forwarding, defaults to false
+`openshift_logging_fluentd_remote_syslog_host`: Required, hostname or IP of remote syslog server
+`openshift_logging_fluentd_remote_syslog_port`: Port of remote syslog server, defaults to 514
+`openshift_logging_fluentd_remote_syslog_severity`: Syslog severity level, defaults to debug
+`openshift_logging_fluentd_remote_syslog_facility`: Syslog facility, defaults to local0
+`openshift_logging_fluentd_remote_syslog_remove_tag_prefix`: Remove the prefix from the tag, defaults to ''
+`openshift_logging_fluentd_remote_syslog_tag_key`: If string specified, use this field from the record to set the key field on the syslog message
+`openshift_logging_fluentd_remote_syslog_use_record`: Set true to use the severity and facility from the record, defaults to false
+`openshift_logging_fluentd_remote_syslog_payload_key`: If string is specified, use this field from the record as the payload on the syslog message
```
Comment (richm) [inline roles/openshift_logging/README.md]: Add another line at the end like "There are corresponding `openshift_logging_mux_` parameters to the above for mux."

File: roles/openshift_logging_fluentd/templates/fluentd.j2.orig [hunk 5]
```diff
@@ -0,0 +1,209 @@
+apiVersion: extensions/v1beta1
```
Comment (richm) [inline roles/openshift_logging_fluentd/templates/fluentd.j2.orig]: I think you need to remove this one from the commit.

**Comment by sqtran** [comment]
Hmm, is this something I did?

> [Fail] [Conformance][Area:Networking][Feature:Router] openshift router metrics The HAProxy router [It] should expose prometheus metrics for a route [Suite:openshift/conformance/parallel] 
> /go/src/github.com/openshift/origin/test/extended/router/metrics.go:128
>

**Comment by richm** [comment]
@sqtran I don't see how - @sdodson is this a flake of some sort?

**Comment by sdodson** [comment]
Yeah it's a flake, https://github.com/openshift/origin/issues/16369

**Comment by sdodson** [comment]
@jcantrill please add kind/bug label to this if you want this to merge now

**Comment by jcantrill** [comment]
[(5) Include and support fluentd-syslog-plugin[logging]](https://trello.com/c/LCte3O0n/469-5-include-and-support-fluentd-syslog-pluginlogging)

**Pull request merged**